### PR TITLE
Migrate notifications.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/notifications.dart [hunk 1]
```diff
@@ -82,7 +83,7 @@ class _InheritedNotifications extends InheritedWidget {
 
 class NotificationsState extends State<_NotificationsProvider>
     implements NotificationService {
-  OverlayEntry _overlayEntry;
+  OverlayEntry? _overlayEntry;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/notifications.dart:86]: i do not see how to make it late
it is initialized in didChangeDependencies that can be invoked more than once, so we need to check the value first
I am open for ideas.

File: packages/devtools_app/lib/src/shared/notifications.dart [hunk 2]
```diff
@@ -112,7 +113,7 @@ class NotificationsState extends State<_NotificationsProvider>
   /// successfully pushed.
   @override
   bool push(
-    String message, {
+    String? message, {
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/notifications.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Comment by polina-c** [comment]
tested

**Pull request merged**